mapillary_tools/video_data_extraction/extract_video_data.py

```diff
@@ -0,0 +1,167 @@
+                    total=len(files),
+                )
+            )
+
+        return video_metadata_or_errors
+
+    def process_file(self, file: Path) -> VideoMetadataOrError:
+        parsers = make_parsers(file, self.options)
+        points: T.Sequence[geo.Point] = []
+        make = None
+        model = None
+
+        exceptions = []
+        for parser in parsers:
+            log_details = {
+                "filename": file,
+                "parser": parser.parser_label,
+                "source": parser.get_geotag_source_path(),
+            }
+            if not points:
+                try:
+                    points = parser.extract_points()
+                except Exception as e:
+                    exceptions.append({"exception": e})
+                    LOG.debug(
+                        "%(filename)s: Exception from parser %(parser)s while processing source %(source)s: %(e)s",
+                        {**log_details, "e": e},
+                    )
+
+                LOG.debug(
+                    "%(filename)s: %(points)d points extracted by parser %(parser)s from file %(source)s}",
+                    {**log_details, "points": len(points)},
+                )
+
+                try:
+                    points = self._sanitize_points(points)
+                except Exception as e:
+                    exceptions.append({"exception": e})
+                    points = []
+                    LOG.debug(
+                        "%(filename)s: Exception during sanitization %(parser)s while processing source %(source)s: %(e)s",
+                        {**log_details, "e": e},
+                    )
+
+                if points:
+                    if parser.must_rebase_times_to_zero:
+                        points = self._rebase_times(points)
+            if not model:
+                model = parser.extract_model()
+            if not make:
+                make = parser.extract_make()
+
+            parser.cleanup()
+
+        if points:
+            return VideoMetadata(
+                filename=file,
+                filetype=FileType.VIDEO,
+                md5sum=None,
+                points=points,
+                make=make,
+                model=model,
+            )
+        else:
+            return ErrorMetadata(
+                filename=file,
+                error=exceptions[-1]["exception"],
+                filetype=FileType.VIDEO,
+            )
+
+    def _screen_sources_cardinality(self, files: T.Sequence[Path]):
+        if len(files) > 1:
+            for parser_opts in self.options["geotag_sources_options"]:
+                if "pattern" in parser_opts and "%" not in parser_opts["pattern"]:
+                    LOG.error(
+                        "Multiple video files found: Geotag source pattern for source %s must include filename placeholders",
+                        parser_opts["source"],
+                    )
+                    exit()
```

Comment:
Is it the process exit? You may throw an exceptions.MapillaryUserException instead https://github.com/mapillary/mapillary_tools/blob/main/mapillary_tools/exceptions.py#L4